### PR TITLE
Modernise

### PR DESCRIPTION
**Pull request by timja**
- Use context specific sidepanel
- Remove sidepanel on main credentials page
- Display store specific names rather than `Jenkins` for all of them on view credentials page
- Remove `norefresh` from views
- Maintain `manage` in url bar so that breadcrumbs work correctly

_Note: screenshots taken on 2.366, on 2.361 the `one-column` layouts will be a bit wider and not look quite as good_

<table>
    <thead>
    <tr>
        <th>
        <th>Before
        <th>After
    </tr>
    <thead>
    <tr>
        <td>View Credentials
        <td>image
        <td>image
    <tr>
        <td>View domain (one-column)
        <td>image
        <td>image
    </tr>
    <tr>
        <td>View domain (domain modifiable)
        <td>image
        <td>image
   </tr>
    <tr>
        <td>View store (domain not modifiable)
        <td>image
        <td>image
    </tr>
    <tr>
        <td>View store (domain modifiable)
        <td>image
        <td>image
    </tr>
    <tr>
    <td>Add domain
     <td>image
      <td>image
     </tr>
</table>

TODO:
- [x] Remove some duplication between `ViewCredentialsAction` and `CredentialsStoreAction`~
- ~Look at fixing [JENKINS-68791](https://issues.jenkins.io/browse/JENKINS-68791) probably with https://github.com/jenkinsci/jenkins/pull/6659~ - will do separately
- [x] Review folder plugin with its credentials provider
- [x] Before and after screenshots

### CODE REVIEW

File: src/main/resources/com/cloudbees/plugins/credentials/ViewCredentialsAction/index.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
 global=(global)
-storesScopedTo=Stores scoped to <a href=".." class='model-link inside'>{0}</a>
+storesScopedTo=Stores scoped to {0}
```
Comment (daniel-beck) [inline src/main/resources/com/cloudbees/plugins/credentials/ViewCredentialsAction/index.properties:2]: Why?

File: src/main/java/com/cloudbees/plugins/credentials/ViewCredentialsAction.java [hunk 2]
```diff
@@ -58,6 +58,7 @@
 import jenkins.model.ModelObjectWithContextMenu;
 import jenkins.model.TransientActionFactory;
 import org.acegisecurity.Authentication;
+import org.apache.commons.lang.StringUtils;
```
Comment (daniel-beck) [inline src/main/java/com/cloudbees/plugins/credentials/ViewCredentialsAction.java:61]: Unused

File: src/main/resources/com/cloudbees/plugins/credentials/ViewCredentialsAction/index.properties (hunk 1 above)
Comment (timja) [reply src/main/resources/com/cloudbees/plugins/credentials/ViewCredentialsAction/index.properties:2]: It just didn't seem useful, why would you want that there?

I looked at renaming this to be clearer what it was but I didn't come up with anything

Comment (daniel-beck) [reply src/main/resources/com/cloudbees/plugins/credentials/ViewCredentialsAction/index.properties:2]: Good point, I missed that this only ever links to the current object, not to its ancestors (which might have been useful).

**Comment by daniel-beck** [comment]
Looks reasonable overall. Especially thanks for taking care of the confusing sidepanels :)

File: pom.xml [hunk 3]
```diff
@@ -67,7 +67,7 @@
   <properties>
     <changelist>999999-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.340</jenkins.version>
+    <jenkins.version>2.361</jenkins.version>
```
Comment (timja) [inline pom.xml:70]: required for the "Manage Jenkins" breadcrumb to be added.

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsStore.java [hunk 4]
```diff
@@ -537,6 +538,22 @@ public final String getContextDisplayName() {
         }
     }
 
+    /**
+     * Resolves a display name from the Store
+     * @return the display name
+     */
+    public String getDisplayName() {
```
Comment (timja) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsStore.java:545]: Moved code from `CredentialsStoreAction`

**Comment by janfaracik** [comment]
Looks good to me!

---



For this page, how would you feel about moving the 'Add domain' into the app bar and making it a primary button? (like the System Log page)

Screenshot 2022-09-06 at 20 07 02

**Comment by timja** [comment]
Yep will try it, it was looking quite weird with a single item there

**Comment by timja** [comment]
mind taking another look? I'll update screenshots later on

File: pom.xml (hunk 3 above)
Comment (timja) [reply pom.xml:70]: yes tomorrow I will update

**Review by NotMyFault — APPROVED** [review]

**Review by janfaracik — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (timja) [reply pom.xml:70]:
```suggestion
    <jenkins.version>2.361.1</jenkins.version>
```

**Comment by timja** [comment]
@jenkinsci/credentials-plugin-developers possible for a review please?

**Comment by jtnord** [comment]
when I go to add a credential I get a page (2 column) that seems out of place with the new design

Given I am in the "add credential" of a current domain" should the "confgure domain" delete/domain and "add credential" show up here?
image

**Review by jtnord — CHANGES_REQUESTED** [review]
something is definatly wrong with the delete button, but see more feedback above

(is red with 2.688 but not with the LTS).

**Comment by timja** [comment]
all the feedback should be addressed 

the class was renamed in 2.364 I could add both, but some of the pages look a bit funny on a one-column layout without other changes in 2.364 so I think better to just raise the baseline to that.

> When viewing Folder credentials there is no obvious way to guide you where to go. (yes the existing design is shitty) but there was a subtask under the credentials task which highlighted the "folder"

I have potential future plans to remove the stores from that page, add a search bar and create a new create credential experience where you can choose the store / type / context in a nicer page.

but I think out of scope of this PR and this should be good enough for now?

**Comment by jtnord** [comment]
> the class was renamed in 2.364 I could add both, but some of the pages look a bit funny on a one-column layout without other changes in 2.364 so I think better to just raise the baseline to that.

I'm a bit hesitant on bumping the baseline this early in the LTS cycle. (There's some fixes still needed to be written for the icon change) and I would like to avoid branching if possible.

Additionally I thought the pages whilst not perfect in the LTS where still an improvement over the previous code.

Would you be against dropping the version requirement to the LTS and adding the old class name in addition?

**Comment by timja** [comment]
Done

**Review by jtnord — APPROVED** [review]

**Comment by jtnord** [comment]
Thanks @timja

**Comment by ite-klass** [comment]
A more descriptive PR title would have been helpful for the release changelog

**Comment by ite-klass** [comment]
*Modernize UI* would clarify it is about user interface / presentation changes

**Pull request merged**